### PR TITLE
gh #82 Modified Audio l1 test cases as per latest audio.h

### PR DESCRIPTION
**Pull request by ishthiyaqahmed**
Modified Audio L1 test suite as per the latest audio.h interface file.

### CODE REVIEW

File: src/test_l1_dsAudio.c [hunk 1]
```diff
@@ -5057,131 +4778,121 @@ void test_l1_dsAudio_positive_dsSetAudioLevel(void) {
...
+ * |03|Call dsSetAudioAtmosOutputMode() - by looping through the acquired port handles to enable Audio ATMOS output mode | handle = [valid handle], enable = [TRUE] | dsERR_NONE | Atmos mode must be enabled for all supported audio ports |
+ * |04|Call dsSetAudioAtmosOutputMode() - by looping through the acquired port handles to disable Audio ATMOS output mode | handle = [valid handle], enable = [FALSE] | dsERR_NONE | Atmos mode must be enabled for all supported audio ports |
+ * |05|Call dsAudioPortTerm() - Terminate audio ports | | dsERR_NONE |  Termination must be successful |
  * 
  */
-void test_l1_dsAudio_positive_dsGetAudioMaxDB(void) {
+void test_l1_dsAudio_positive_dsSetAudioAtmosOutputMode(void) {
 	// Start of the test
-	gTestID = 81;
+	gTestID = 77; // Assuming Test Case ID is 238
```
Comment (bhanucbp) [inline src/test_l1_dsAudio.c:4869]: Id in comment is wrong. remove the comment. Check other places also

File: src/test_l1_dsAudio.c [hunk 2]
```diff
@@ -10102,10 +8011,6 @@ int test_l1_dsAudio_register ( void )
 	UT_add_test( pSuite, "dsAudioPortTerm_L1_negative" ,test_l1_dsAudio_negative_dsAudioPortTerm );
```
Comment (bhanucbp) [inline src/test_l1_dsAudio.c:8011]: Some APIs are specific to source or sink. If the API is not supported, expected to return NOT_SUPPORTED. Split those test cases by adding _source or _sink postfix and modify accordingly. When yaml changes introduced you can filter the register based on the platform

File: src/test_l1_dsAudio.c [hunk 3]
```diff
@@ -10102,10 +8011,6 @@ int test_l1_dsAudio_register ( void )
...
 	UT_add_test( pSuite, "dsGetAudioPort_L1_positive" ,test_l1_dsAudio_positive_dsGetAudioPort );
 	UT_add_test( pSuite, "dsGetAudioPort_L1_negative" ,test_l1_dsAudio_negative_dsGetAudioPort );
```
Comment (bhanucbp) [inline src/test_l1_dsAudio.c:8013]: test cases for below APIs are missing:

dsAudioEnableARC
dsAudioFormatUpdateRegisterCB
dsAudioSetSAD
dsGetSupportedARCTypes

**Review by bhanucbp — COMMENTED** [review]
address the review comments

File: src/test_l1_dsAudio.c (hunk 2 above)
Comment (thanushreevani) [reply src/test_l1_dsAudio.c:8011]: you mean split the test cases like 
test_l1_dsAudio_positive_dsGetMS12AudioProfileList_source
test_l1_dsAudio_positive_dsGetMS12AudioProfileList_sink

but wont that be a duplicate work of the next task(enhancement of using yaml) that we are going to do?

File: src/test_l1_dsAudio.c (hunk 1 above)
Comment (thanushreevani) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c (hunk 3 above)
Comment (thanushreevani) [reply src/test_l1_dsAudio.c:8484]: Done

Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c:8484]: hi bhanu,
While developing L1 test cases we agreed to move the test cases to L2 which has dependency of external devices.
Do we need to add it again ?

File: src/test_l1_dsAudio.c [hunk 4]
```diff
@@ -8219,1674 +7789,600 @@ void test_l1_dsAudio_positive_dsGetAudioCapabilities(void) {
...
 
-		// Set Secondary language
-		result = dsSetSecondaryLanguage(handle[i], "ENG");
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
+		if(handle[i] == dsAUDIOPORT_TYPE_HDMI_ARC) {
+			// Step 03: supported ARC types of the connected ARC/eARC
+			result = dsGetSupportedARCTypes(handle[i], &types[i]);
+			UT_ASSERT_EQUAL(result, dsERR_NONE);
+		}
+		else {
```
Comment (bhanucbp) [inline src/test_l1_dsAudio.c:8125]: else part is not required

File: src/test_l1_dsAudio.c [hunk 5]
```diff
@@ -8219,1674 +7789,600 @@ void test_l1_dsAudio_positive_dsGetAudioCapabilities(void) {
...
 		UT_ASSERT_EQUAL(result, dsERR_NONE);
 		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
 
-		// Set Secondary language
-		result = dsSetSecondaryLanguage(handle[i], "ENG");
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
+		if(handle[i] == dsAUDIOPORT_TYPE_HDMI_ARC) {
+			// Step 03: supported ARC types of the connected ARC/eARC
+			result = dsGetSupportedARCTypes(handle[i], &types[i]);
+			UT_ASSERT_EQUAL(result, dsERR_NONE);
```
Comment (bhanucbp) [inline src/test_l1_dsAudio.c:8123]: this API might not return dsERR_NONE as ARC port is not connected. Check with appropriate error code

File: src/test_l1_dsAudio.c [hunk 6]
```diff
@@ -8219,1674 +7789,600 @@ void test_l1_dsAudio_positive_dsGetAudioCapabilities(void) {
...
 
-		// Step 05: Call with NULL language pointer
-		result = dsSetSecondaryLanguage(handle[i], NULL);
-		UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
-
-		// Step 06: Call with unsupported language code
-		result = dsSetSecondaryLanguage(handle[i], "XYZ");
-		UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
+		// Step 05: Attempt to get SupportedARCTypes with a null pointer
+			result = dsGetSupportedARCTypes(handle[i], NULL);
```
Comment (bhanucbp) [inline src/test_l1_dsAudio.c:8189]: call only for ARC port handle
cosmetic: replace tabs with 4 spaces

**Review by bhanucbp — COMMENTED** [review]
please address review comments

File: src/test_l1_dsAudio.c (hunk 5 above)
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c:8123]: as per the code in llama it is returning dsERR_NONE for non supported ports , need hal changes for the not supported ports , for now will remove the else part.

File: src/test_l1_dsAudio.c [hunk 7]
```diff
@@ -6933,116 +6563,98 @@ void test_l1_dsAudio_positive_dsEnableLEConfig(void) {
...
+	gTestID = 106; // Assuming Test Case ID is 106
 	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
 
 	int result;
-	intptr_t  handle[NUM_OF_PORTS] = {INT_ARRAY_INIT};
 
-	// Step 01: Attempt to enable LE without initializing
-	result = dsEnableLEConfig(-1, true);
+	// Step 01: Attempt to register callback without initializing ports
+	result = dsAudioFormatUpdateRegisterCB(NULL);
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 8]
```diff
@@ -8219,1674 +7789,600 @@ void test_l1_dsAudio_positive_dsGetAudioCapabilities(void) {
...
-		result = dsGetSecondaryLanguage(handle[i], languageCode1);
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
-
-		// Step 05 : Get Secondary language in a new array
-		UT_ASSERT_EQUAL(languageCode, languageCode1);
+		// Step 03: Set SAD for HDMI ARC port
+		if(handle[i] == dsAUDIOPORT_TYPE_HDMI_ARC)
+		{
+			result = dsAudioSetSAD(handle[i], sadlist);
+			UT_ASSERT_EQUAL(result, dsERR_NONE);
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c:8254]: as per the hal code it is returning dsERR_NONE for non supported ports , think need hal changes to return not supported code.

File: src/test_l1_dsAudio.c [hunk 9]
```diff
@@ -8219,1674 +7789,600 @@ void test_l1_dsAudio_positive_dsGetAudioCapabilities(void) {
...
 		UT_ASSERT_EQUAL(result, dsERR_NONE);
 		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
 
-		// Step 05: Call with NULL language pointer
-		result = dsGetSecondaryLanguage(handle[0], NULL);
+		// Step 05: Attempt to set Short Audio Descriptor with an invalid sad value
+		dsAudioSADList_t sadlist1;
+	    sadlist1.sad[0] = -1; sadlist1.sad[1] = -2; sadlist1.sad[2] = -3; sadlist1.sad[3] = -4;
+	    sadlist1.count = 20; //max sad is 15
+		result = dsAudioSetSAD(handle[i], sadlist1);
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 10]
```diff
@@ -8219,1674 +7789,600 @@ void test_l1_dsAudio_positive_dsGetAudioCapabilities(void) {
...
 	for (int i = 0; i < NUM_OF_PORTS; i++) {
 		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
 		UT_ASSERT_EQUAL(result, dsERR_NONE);
 		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
 
-		// Step 05: Call with NULL language pointer
-		result = dsGetSecondaryLanguage(handle[0], NULL);
+		// Step 05: Attempt to set Short Audio Descriptor with an invalid sad value
+		dsAudioSADList_t sadlist1;
+	    sadlist1.sad[0] = -1; sadlist1.sad[1] = -2; sadlist1.sad[2] = -3; sadlist1.sad[3] = -4;
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c (hunk 4 above)
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c (hunk 6 above)
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

**Comment by thanushreevani** [comment]
No comments

**Review by thanushreevani — DISMISSED** [review]

**Review by bhanucbp — APPROVED** [review]
ship it

File: skeletons/src/dsAudio.c [hunk 11]
```diff
@@ -565,6 +565,13 @@ dsError_t dsAudioFormatUpdateRegisterCB(dsAudioFormatUpdateCB_t cbFun)
   return (dsError_t)0;
```
Comment (hari22yuva) [inline skeletons/src/dsAudio.c:565]: please remove the stub implementation from `skeletons/src/dsAudio.c` which is removed from the header

Comment (ishthiyaqahmed) [reply skeletons/src/dsAudio.c:475]: done

**Comment by hari22yuva** [comment]
> _ No description provided. _

Please add description

**Comment by ishthiyaqahmed** [comment]
> > _ No description provided. _
> 
> Please add description


done

**Review by hari22yuva — APPROVED** [review]
approved

**Pull request merged**